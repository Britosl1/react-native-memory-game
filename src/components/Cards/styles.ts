import styled, {css} from 'styled-components/native';

type CardStyleProps = {
  flipped: boolean;
};

const cardStyle = css`
  border-radius: 5px;
  width: 60px;
  height: 60px;
  margin: 5px;
  align-items: center;
  justify-content: center;
`;

export const FrontCard = styled.Image<CardStyleProps>`
  ${cardStyle}
  z-index: ${props => (props.flipped ? 2 : 1)};
  transform: ${props => (props.flipped ? 'rotate(0deg)' : 'rotateY(180deg)')};
`;

export const BackCard = styled.Image<CardStyleProps>`
  ${cardStyle}
  z-index: ${props => (props.flipped ? 1 : 2)};
  transform: ${props => (props.flipped ? 'rotateY(180deg)' : 'rotate(360deg)')};
  position: absolute;
  top: 0px;
  left: 0px;
`;
