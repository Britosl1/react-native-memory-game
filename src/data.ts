import Card1 from '../src/images/1.png';
import Card2 from '../src/images/2.png';
import Card3 from '../src/images/3.png';
import Card4 from '../src/images/4.png';
import Card5 from '../src/images/5.png';
import Card6 from '../src/images/6.png';
import Card7 from '../src/images/7.png';
import Card8 from '../src/images/8.png';
import Card9 from '../src/images/9.png';
import Card10 from '../src/images/10.png';
import Back from '../src/images/back.png';

import {ImageSourcePropType} from 'react-native';

export type DataType = {
  id: string;
  img: ImageSourcePropType;
  backImg: ImageSourcePropType;
  flipped: boolean;
  clickable: boolean;
  matchingCardId: string;
};

const cards: ImageSourcePropType[] = [
  Card1,
  Card2,
  Card3,
  Card4,
  Card5,
  Card6,
  Card7,
  Card8,
  Card9,
  Card10,
];
export const data: DataType[] = [...cards, ...cards].map((card, i) => ({
  id: `card${i}`,
  flipped: false,
  img: card,
  backImg: Back,
  clickable: true,
  matchingCardId:
    i < cards.length ? `card${i + cards.length}` : `card${i - cards.length}`,
}));
