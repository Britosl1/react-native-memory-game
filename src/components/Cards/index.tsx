import React from 'react';
import {TouchableOpacity} from 'react-native';
import {FrontCard, BackCard} from './styles';
import {DataType} from '../../data';

interface ICardsProps {
  card: DataType;
  callback: (card: DataType) => void;
}

const Cards: React.FC<ICardsProps> = ({card, callback}) => {
  const handleClick = () => {
    if (card.clickable) {
      callback(card);
    }
  };

  return (
    <TouchableOpacity onPress={handleClick}>
      <FrontCard source={card.img} flipped={card.flipped} />
      <BackCard source={card.backImg} flipped={card.flipped} />
    </TouchableOpacity>
  );
};

export default Cards;
