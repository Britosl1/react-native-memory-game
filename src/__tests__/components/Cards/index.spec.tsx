import React from 'react';
import {render} from '@testing-library/react-native';
import Cards from '../../../components/Cards';
import {DataType} from '../../../data';
import Img from '../../../images/1.png';

describe('Snapshot test', () => {
  test('Card component', () => {
    const data: DataType = {
      id: '1',
      clickable: false,
      flipped: false,
      matchingCardId: '1',
      backImg: Img,
      img: Img,
    };
    const {toJSON} = render(<Cards callback={jest.fn()} card={data} />);

    expect(toJSON()).toMatchSnapshot();
  });
});
