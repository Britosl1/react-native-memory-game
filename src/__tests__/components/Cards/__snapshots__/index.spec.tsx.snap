// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test Card component 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Image
    flipped={false}
    source={
      Object {
        "testUri": "../../../src/images/1.png",
      }
    }
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 5,
          "height": 60,
          "justifyContent": "center",
          "marginBottom": 5,
          "marginLeft": 5,
          "marginRight": 5,
          "marginTop": 5,
          "transform": Array [
            Object {
              "rotateY": "180deg",
            },
          ],
          "width": 60,
          "zIndex": 1,
        },
      ]
    }
  />
  <Image
    flipped={false}
    source={
      Object {
        "testUri": "../../../src/images/1.png",
      }
    }
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 5,
          "height": 60,
          "justifyContent": "center",
          "left": 0,
          "marginBottom": 5,
          "marginLeft": 5,
          "marginRight": 5,
          "marginTop": 5,
          "position": "absolute",
          "top": 0,
          "transform": Array [
            Object {
              "rotate": "360deg",
            },
          ],
          "width": 60,
          "zIndex": 2,
        },
      ]
    }
  />
</View>
`;
